lib/adagrams.rb

```diff
@@ -0,0 +1,171 @@
+require 'pry'
+# require_relative '../assets/dictionary-english.csv' #for wave 5 
+
+# wave 1
+
+# draw_letters includes letter library due to scope since method is instructed to not require parameters
+def draw_letters
+  letters = [
+    { character: "a", quantity: 9, score: 1 },
+    { character: "b", quantity: 2, score: 3 },
+    { character: "c", quantity: 2, score: 3 },
+    { character: "d", quantity: 4, score: 2 },
+    { character: "e", quantity: 12, score: 1 },
+    { character: "f", quantity: 2, score: 4 },
+    { character: "g", quantity: 3, score: 2 },
+    { character: "h", quantity: 2, score: 4 },
+    { character: "i", quantity: 9, score: 1 },
+    { character: "j", quantity: 1, score: 8 },
+    { character: "k", quantity: 1, score: 5 },
+    { character: "l", quantity: 4, score: 1 },
+    { character: "m", quantity: 2, score: 3 },
+    { character: "n", quantity: 6, score: 1 },
+    { character: "o", quantity: 8, score: 1 },
+    { character: "p", quantity: 2, score: 3 },
+    { character: "q", quantity: 1, score: 10 },
+    { character: "r", quantity: 6, score: 1 },
+    { character: "s", quantity: 4, score: 1 },
+    { character: "t", quantity: 6, score: 1 },
+    { character: "u", quantity: 4, score: 1 },
+    { character: "v", quantity: 2, score: 4 },
+    { character: "w", quantity: 2, score: 4 },
+    { character: "x", quantity: 1, score: 8 },
+    { character: "y", quantity: 2, score: 4 },
+    { character: "z", quantity: 1, score: 10 },
+  ]
+  # letter_bucket creates an empty array that will be used to have a representative (gigantic) alphabet library
+  letter_bucket = []
+  letters.each do |letter|
+    letter[:quantity].times do
+      letter_bucket << letter[:character]
+    end
+  end
+  # hand creates an empty array that will have 10 randomly drawn letters
+  hand = []
+  10.times do 
+    hand << letter_bucket.slice(rand(0...letter_bucket.length)) 
+  end
+  # restores the "used" letters to letter_bucket so they are not "used up"
+  letter_bucket << hand
+  letter_bucket.flatten!.sort!
+  return hand
+end
+
+# wave 2 
+# iterate over user-provided input and hand dealt
+# as the characters match, they are removed from each collection
+def uses_available_letters?(input, letters_in_hand)
+  input = input.split('')
+  letters_in_hand.reject do |letter|
+    if input.include?(letter)
+      input.slice!(input.index(letter))
+    end
+  end
+  # the length of the input checks that the user used only available letters
+  if input.length == 0
+    return true
+  else
+    return false
+  end
+end
+
+# wave 3
+
+# manipulate user input
+def score_word (word)
```

Comment:
Minor nitpick: You CAN define a method with a space between the method name and the input parens, but Ruby will give a warning, because it prefers it if the method signature looks more like `def score_word(word)`